#!/usr/bin/env python
PACKAGE = "asoc_upper_controller"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
gen.add("para_vel", double_t, 0, "para_vel", 5.0, 0, 50.0)
gen.add("P_Long", double_t, 0, "Longitude_speed", 9.5, 0, 50.0)
gen.add("D_long", double_t, 0, "longitude damp", 2.0, 0, 50)
gen.add("P_Yaw", double_t, 0, "P for yaw rate", 27.0, 0, 80.0)
gen.add("D_Yaw", double_t, 0, "D for yaw rate", 25.0, 0, 80.0)
gen.add("forward_dist", double_t, 0, "forward_dist", 0.1, 0, 2.0)
gen.add("forward_dist2", double_t, 0, "forward_dist2", 0.1, 0, 2.0)
gen.add("forward_dist3", double_t, 0, "forward_dist3", 0.1, 0, 2.0)
gen.add("P_Lateral", double_t, 0, "P for lateral dist", 40.0, 0, 80.0)
gen.add("D_Lateral", double_t, 0, "D for lateral dist", 20.0, 0, 80.0)
gen.add("P_pit", double_t, 0, "P for pitch", 0, 0, 300.0)
gen.add("D_pit", double_t, 0, "D for pitch", 0, 0, 50.0)
gen.add("P_rol", double_t, 0, "P for roll", 0, 0, 300.0)
gen.add("D_rol", double_t, 0, "D for roll", 0, 0, 50.0)
gen.add("slow_ff", double_t, 0, "factor inside the slow factor", 1.0, 0, 5.0)




exit(gen.generate(PACKAGE, "asoc_upper_controller", "controller_"))
